pages/guides/getting-started/remix-guide.mdx

```diff
@@ -14,66 +14,208 @@ Inside your `Remix` project root directory, install Chakra UI by running either
+client and the server. We'll use our createEmotionCache function here.
 
 ```jsx live=false
-export function ErrorBoundary({ error }: { error: Error }) {
+// entry.client.tsx
+import * as React from 'react'
+import { hydrate } from 'react-dom'
+import { CacheProvider } from '@emotion/react'
+import { RemixBrowser } from 'remix'
+
+import { ClientStyleContext } from './context'
+import { createEmotionCache } from './createEmotionCache'
+
+interface ClientCacheProviderProps {
+  children: React.ReactNode;
+}
+
+function ClientCacheProvider({ children }: ClientCacheProviderProps) {
+  const [cache, setCache] = React.useState(createEmotionCache())
+
+  function reset() {
+    setCache(createEmotionCache())
+  }
+
   return (
-    <Document title='Error!'>
-      <ChakraProvider>
-        <Box>
-          <Heading as='h1'>There was an error</Heading>
-        </Box>
-      </ChakraProvider>
-    </Document>
+    <ClientStyleContext.Provider value={{ reset }}>
+      <CacheProvider value={cache}>{children}</CacheProvider>
+    </ClientStyleContext.Provider>
   )
 }
 
-export function CatchBoundary() {
-  let caught = useCatch()
+hydrate(
+  <ClientCacheProvider>
+    <RemixBrowser />
+  </ClientCacheProvider>,
+  document,
+)
+```
+
+```jsx live=false
+// entry.server.tsx
+import { renderToString } from 'react-dom/server'
+import { CacheProvider } from '@emotion/react'
+import createEmotionServer from '@emotion/server/create-instance'
+import type { EntryContext } from 'remix'
+import { RemixServer } from 'remix'
+import 'dotenv/config'
+
+import { ServerStyleContext } from './context'
+import { createEmotionCache } from './createEmotionCache'
+
+export default function handleRequest(
+  request: Request,
+  responseStatusCode: number,
+  responseHeaders: Headers,
+  remixContext: EntryContext,
+) {
+  const cache = createEmotionCache()
+  const { extractCriticalToChunks } = createEmotionServer(cache)
+
+  const html = renderToString(
+    <ServerStyleContext.Provider value={null}>
+      <CacheProvider value={cache}>
+        <RemixServer context={remixContext} url={request.url} />
+      </CacheProvider>
+    </ServerStyleContext.Provider>,
+  )
+
+  const chunks = extractCriticalToChunks(html)
+
+  const markup = renderToString(
+    <ServerStyleContext.Provider value={chunks.styles}>
+      <CacheProvider value={cache}>
+        <RemixServer context={remixContext} url={request.url} />
+      </CacheProvider>
+    </ServerStyleContext.Provider>,
+  )
+
+  responseHeaders.set('Content-Type', 'text/html')
+
+  return new Response(`<!DOCTYPE html>${markup}`, {
+    status: responseStatusCode,
+    headers: responseHeaders,
+  })
+}
+```
 
+Inside our root.tsx file we'll create a `Document` wrapper and then we'll wrap
+our `App` with the Document.
+
+```jsx live=false
+// root.tsx
+interface DocumentProps {
+  children: React.ReactNode;
+}
+
+const Document = withEmotionCache(
```

Comment:
You're missing import statements for:
- `withEmotionCache`
- `React`
- `ServerStyleContext`
- `ClientStyleContext`
- `ChakraProvider`